src/scenes/data/components/activityList/index.jsx

```diff
@@ -40,16 +51,34 @@ function ActivityList(props) {
         return `${numHours}H ${numMinutes}M`;
     }
 
+    function noPremiumOnHover(state, activity) {
+        if (state !== 'premium' && !activity.inActivities) {
+            setHover(activity.name);
+        }
+    }
+
+    function noPremiumOffHover(state, activity) {
+        if (state !== 'premium' && !activity.inActivities) {
+            setHover(null);
+        }
+    }
+
```

Comment:
These two functions could be consolidated into one.

```suggestion
    function handleHover(state, acitvity) {
        if (state !== 'premium' && !activity.inActivities) {
            setHover(hover ? null : activity.name);
        }
    }
```